### ISSUE TITLE
weekNumberInMonth returning incorrect week number

### ISSUE THREAD

**Issue by Chris-Pratt-Clystnet** [briannesbitt/Carbon#1386]
So ran into some problems with this function.

When the function got the week number for this month on Monday it got 1 rather than 2.

Looking at the code you have 
```php
return (int) ceil(($this->day + $this->copy()->startOfMonth()->dayOfWeek - 1) / static::DAYS_PER_WEEK);
```
but that calculation falls down in this instance as (9 + 0 - 1) / 7 = 1.143.

**Comment by 36864** [comment]
The call to `ceil()` would make that 2.

Can you share the code where the error happened? It's more likely that a timezone issue caused your monday script to technically run on sunday.

**Comment by kylekatarnls** [comment]
I can confirm the bug:
```php
echo Carbon::createMidnightDate(2018, 7, 2, 'UTC')->weekNumberInMonth;
```
Returns `1` and I agree it seems logical to get `2`. (Tested with version 1.32.0 in http://try-carbon.herokuapp.com/)

But I have to properly check the expectation for this property.

**Comment by kylekatarnls** [comment]
It seems to me that we should use `dayOfWeekIso` (with Sunday = 7) rather than `dayOfWeek` in this calcul, I will check.

**Comment by 36864** [comment]
Can't reproduce. Using dayOfWeek resolves to week 2, using dayOfWeekIso resolves to week 3:
```
>>> $now = now();
=> Illuminate\Support\Carbon @1531094400 {#2848
     date: 2018-07-09 00:00:00.0 UTC (+00:00),
   }
>>> (int) ceil(($now->day + $now->copy()->startOfMonth()->dayOfWeek) / Carbon::DAYS_PER_WEEK)
=> 2
>>> (int) ceil(($now->day + $now->copy()->startOfMonth()->dayOfWeekIso) / Carbon::DAYS_PER_WEEK)
=> 3
>>>   
```

**Comment by kylekatarnls** [comment]
Yes, and 3 is right considering the first week is a 1-sunday-day week.

**Comment by kylekatarnls** [comment]
Fixed in version 2 #1385 and will be copied in 1.33

**Comment by Chris-Pratt-Clystnet** [comment]
Closing this ticket as the fix you have implemented is exactly what i needed so thanks :-)
